src/adapters/MorphoMarketV1Adapter.sol

```diff
@@ -94,6 +97,16 @@ contract MorphoMarketV1Adapter is IMorphoMarketV1Adapter {
         require(msg.sender == parentVault, NotAuthorized());
         require(marketParams.loanToken == asset, LoanAssetMismatch());
 
+        if (shares[marketId] == 0 && assets > 0) {
+            for (uint256 i = 0; i < allMarketParams.length; i++) {
```

Comment:
The allocator could pass the index `i` in the data. This is also compatible with the liquidity adapter, as it could be in the liquidityData as well

Reply:
There is the annoying part that if a market is removed, you may need to update the liquidityData

Reply:
yes, exactly for this reason I thought that it was better this way

Reply:
note that the added gas is not much, because most of the time you don't create/empty a market